python/foglamp/services/core/api/north.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+from aiohttp import web
+
+from foglamp.services.core import server
+from foglamp.common.storage_client.payload_builder import PayloadBuilder
+from foglamp.services.core import connect
+
+
+__author__ = "Praveen Garg"
+__copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_help = """
+    -------------------------------------------------------------------------------
+    | GET                 | /foglamp/north                                        |
+    -------------------------------------------------------------------------------
+"""
+
+
+async def _get_sent_stats():
+    stats = []
+    try:
+        payload = PayloadBuilder().SELECT("key", "value").payload()
+        storage_client = connect.get_storage_async()
+        result = await storage_client.query_tbl_with_payload('statistics', payload)
+        if int(result['count']):
+            stats = result['rows']
+    except:
+        raise
+    else:
+        return stats
+
+
+async def _get_north_schedules():
+    schedules = []
+
+    schedule_list = await server.Server.scheduler.get_schedules()
+    for sch in schedule_list:
+        if sch.group.upper() == 'NORTH':
+            schedules.append({
+                'id': str(sch.schedule_id),
+                'name': sch.name,
+                'processName': sch.process_name,
+                'repeat': sch.repeat.total_seconds() if sch.repeat else 0,
+                'day': sch.day,
+                'enabled': sch.enabled,
+                'exclusive': sch.exclusive
+            })
+
+    return schedules
+
+
+async def get_north_schedules(request):
+    """
+    Args:
+        request:
+
+    Returns:
+            list of all schedules for processes of type 'tasks/north' or 'tasks/north_c'
+
+    :Example:
+            curl -X GET http://localhost:8081/foglamp/north
+    """
+    try:
+        north_schedules = await _get_north_schedules()
+        stats = await _get_sent_stats()
+
+        for sch in north_schedules:
+            # FIXME: sch.name vs sch.processName?
+            stat = next((s for s in stats if s["key"] == sch["name"]), None)
```

Comment:
Safe to use sch["processName"], with a TODO, for now as this is due to a bug in the scheduler code which will be removed in a near future sprint.